pkg/purl/purl.go

```diff
@@ -238,6 +297,32 @@ func purlType(t string) string {
 	return t
 }
 
+func TypeFromLibraryComponent(c cdx.Component) (string, error) {
```

Comment:
We should not have CycloneDX-specific logic in the `purl` package.

Reply:
move to cyclonedx pkg.